cpp/ql/lib/semmle/code/cpp/dataflow/ExternalFlow.qll

```diff
@@ -367,16 +400,154 @@ private predicate elementSpec(
   summaryModel(namespace, type, subtypes, name, signature, ext, _, _, _, _)
 }
 
-private string paramsStringPart(Function c, int i) {
-  i = -1 and result = "(" and exists(c)
+/** Gets the fully templated version of `f`. */
```

Comment:
Yes, exactly. This is harder than it sounds because we also need the function belonging to a _class_ template. For example:
```cpp
template<typename T>
class C {
  void f(const T& t);
};
``` 
the `f` is not a function template, but it is a member function of the enclosing class template. And we need to find _that_ function.